extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/ResteasyBuiltinsProcessor.java

```diff
@@ -51,7 +51,7 @@ void setUpDenyAllJaxRs(CombinedIndexBuildItem index,
             JaxRsSecurityConfig config,
             ResteasyDeploymentBuildItem resteasyDeployment,
             BuildProducer<AdditionalSecuredClassesBuildItem> additionalSecuredClasses) {
-        if (config.denyJaxRs) {
+        if ((config.denyJaxRs) && (resteasyDeployment != null)) {
```

Comment:
Isn’t the build item mandatory in this case? I’m surprised you don’t have an error instead of a null value.

Reply:
@stuartwdouglas is it normal that an injected build item can be null? I thought we needed to use Optional in this case to avoid dependency injection issues?

Reply:
I don't remember what the behavior is, but I can certainly look into it if this becomes an issue